web/src/main/java/org/fao/geonet/kernel/harvest/harvester/csw/Aligner.java

```diff
@@ -170,7 +176,30 @@ public CswResult align(Set<RecordInfo> records) throws Exception
 	//---
 	//--------------------------------------------------------------------------
 
-	private void addMetadata(RecordInfo ri) throws Exception
+    private void parseXSLFilter() {
+        processName = params.xslfilter;
+        
+        // Parse complex xslfilter process_name?process_param1=value&process_param2=value...
+        if (params.xslfilter.contains("?")) {
+            String[] filterInfo = params.xslfilter.split("\\?");
```

Comment:
Is there nothing better available to parse key-value pairs out of urls in geonetwork than handwriting the code?